Fix testing now that random.sample() arg must be a sequence

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #94056
* __->__ #94052
* #94051

This is only enforced in 3.11 but the change is not bad for other versions either (and this is test code so perf is not a concern).

albanD committed:
Fix testing now that random.sample() arg must be a sequence

[ghstack-poisoned]

Skylion007 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -5744,7 +5744,7 @@ def sample_inputs_cross_entropy(op_info, device, dtype, requires_grad, **kwargs)
             if "ignore_index" in kwargs and torch.all(target == kwargs["ignore_index"]):
                 # make sure at least one item in target is not ignored
-                target[0] = random.sample(set(range(num_classes)) - {kwargs["ignore_index"]}, 1)[0]
+                target[0] = random.sample(sorted(set(range(num_classes)) - {kwargs["ignore_index"]}), 1)[0]
```
why sorted versus list()? extra overhead and it doesn't buy you any randomness :P

albanD replied:
Because that's what the Python god recommended in the error message:
```
  File "/home/albandes/local/pytorch/3.11_debug_source/torch/testing/_internal/common_methods_invocations.py", line 5747, in sample_inputs_cross_entropy
    target[0] = random.sample(set(range(num_classes)) - {kwargs["ignore_index"]}, 1)[0]
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/albandes/local/installs/python3.11/debug/install/lib/python3.11/random.py", line 439, in sample
    raise TypeError("Population must be a sequence.  "
TypeError: Population must be a sequence.  For dicts or sets, use sorted(d).
```

albanD committed:
Update on "Fix testing now that random.sample() arg must be a sequence"


This is only enforced in 3.11 but the change is not bad for other versions either (and this is test code so perf is not a concern).

[ghstack-poisoned]

albanD commented:
@pytorchbot merge
